odoo/addons/base/models/res_bank.py

```diff
@@ -96,6 +97,12 @@ def _compute_sanitized_acc_number(self):
         for bank in self:
             bank.sanitized_acc_number = sanitize_account_number(bank.acc_number)
 
+    @api.constrains('acc_number', 'sanitized_acc_number')
+    def _check_sanitized_acc_number(self):
+        for bank in self:
+            if bank.sanitized_acc_number != sanitize_account_number(bank.acc_number):
+                raise ValidationError(_('The sanitized account number is invalid.'))
```

Comment:
since `_sanitize_vals` is done when `create` and `write`, is this `constraints` still necessary?

Reply:
Meh, it is a constraint that makes sense functionally; I just kept it because I didn't want to rely on the implementation to be sure it always holds.

I removed it... We'll rely on the tests I guess (which are not testing exactly this)